dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,192 @@
+import logging
+from datetime import datetime
+from typing import Any, Dict, List
+
+from analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+from analyzer_helper.discord.utils.is_user_bot import UserBotChecker
+
+
+class DiscordTransformRawData(TransformRawDataBase):
+    def __init__(self, platform_id: str):
+        """
+        Initializes the class with platform and bot checker.
+        """
+        self.platform_id = platform_id
+        self.user_bot_checker = UserBotChecker(platform_id)
+
+    def create_interaction_base(
+        self,
+        name: str,
+        users_engaged_id: List[str],
+        type: str,
+    ) -> Dict[str, Any]:
+        """
+        Creates an interaction dictionary.
+
+        Args:
+            interaction_type (str): Name of the interaction (e.g., 'reply', 'mention').
+            users_engaged_id (List[str]): List of user IDs engaged in the interaction.
+            type (str): Type of the interaction (e.g., 'emitter', 'receiver').
+
+        Returns:
+            Dict[str, str]: Dictionary representing the interaction.
+        """
+        return {
+            "name": name,
+            "users_engaged_id": users_engaged_id,
+            "type": type,
+        }
+
+    def create_interaction(
+        self,
+        data: Dict[str, Any],
+        name: str,
+        author: str,
+        engaged_users: List[str],
+        type: str,
+    ) -> Dict[str, Any]:
+        """
+        Creates an interaction dictionary.
+
+        Args:
+            data (Dict[str, Any]): Raw data containing interaction details.
+            name (str): Name of the interaction (e.g., 'reply', 'mention').
+            author (str): ID of the author of the interaction.
+            engaged_users (List[str]): List of IDs of the engaged users.
+            type (str): Type of the interaction (e.g., 'receiver', 'emitter').
+
+        Returns:
+            Dict[str, Any]: Dictionary representing the interaction.
+        """
+        is_bot = self.user_bot_checker.is_user_bot(author)
+        return {
+            "author_id": author,
+            "date": data.get("createdDate", data.get("period")),
+            "source_id": data["messageId"],
+            "metadata": {
+                "thread_id": data["threadId"],
+                "channel_id": data["channelId"],
+                "bot_activity": data["isGeneratedByWebhook"] or is_bot,
+            },
+            "actions": [],
+            "interactions": [
+                self.create_interaction_base(
+                    name=name, users_engaged_id=engaged_users, type=type
+                )
+            ],
+        }
+
+    def create_transformed_item(self, data, period, interactions):
```

Comment:
type annotation missing for function input parameters. Please consider adding them. Also, the `period` is not representative of the message date. Consider getting it from `createdDate` field of data.